### PR TITLE
Fix valve-firmware

### PR DESCRIPTION
**Pull request by LukeShortCloud**
3 somewhat changes in this PR:
1. `valve-firmware` now works again. `linux-firmware` was recently updated in Fedora adding in conflicting firmware files. This fixes unstable OS builds.
    - Tested and working on my OLED Steam Deck. Wi-Fi still works.
3. Fedora Copr now has Feodra 41 pipelines enabled. GitHub Actions pipelines have been updated to account for that.
4. The unused Linux kernel build has been removed since we are using `kernel-fsync` instead.

### CODE REVIEW

File: valve-firmware/valve-firmware.spec [hunk 1]
```diff
@@ -36,11 +36,7 @@ mv usr/lib/firmware/cs35l41-dsp1-spk-{cali.bin,cali.wmfw}.xz %{buildroot}/%{_pre
 tar -xvf %{_sourcedir}/steamdeck-dsp-*-any.pkg.tar.zst -C %{buildroot}/
 rm -f %{buildroot}/.BUILDINFO %{buildroot}/.MTREE %{buildroot}/.PKGINFO %{buildroot}/etc/wireplumber
 
-# Fix Steam Deck OLED Wi-Fi firmware with Linux >= 6.9 kernels.
-ln -s QCA206X %{buildroot}/usr/lib/firmware/ath11k/QCA2066
```
Comment (alkazar) [inline valve-firmware/valve-firmware.spec:40]: Wouldn't this still be need for Fedora 40 and 39?

Comment (LukeShortCloud) [reply valve-firmware/valve-firmware.spec:40]: Good point, thanks for bringing this up. So here's the odd situation for the latest `linux-firmware` package update:

- Fedora 41 = In testing repository
- Fedora 40 = In stable repository
- Fedora 39 = In testing repository

We should wait to merge this until it's been promoted to the stable repository (at least for Fedora 39).

**Review by alkazar — APPROVED** [review]

File: valve-firmware/valve-firmware.spec (hunk 1 above)
Comment (LukeShortCloud) [reply valve-firmware/valve-firmware.spec:40]: Fedora 39 has now promoted the new `linux-firmware` package to the stable updates repository. Fedora 41 has not but, considering it's a beta (and the updated package is staged in the testing repository), I think it's fine for us to merge this now.

- https://mirror.rackspace.com/fedora/updates/39/Everything/x86_64/Packages/l/
- https://mirror.rackspace.com/fedora/updates/testing/41/Everything/x86_64/Packages/l/

**Comment by LukeShortCloud** [comment]
The Fedora Copr build worked for Fedora 39 and 40 but not 41. I'm not sure why, but it's trying to create a debug package.

```
+ /usr/bin/add-determinism --brp -j2 /builddir/build/BUILD/valve-firmware-20231113.1-build/BUILDROOT
Scanned 39 directories and 53 files,
               processed 0 inodes,
               0 modified (0 replaced + 0 rewritten),
               0 unsupported format, 0 errors
Reading /builddir/build/BUILD/valve-firmware-20231113.1-build/SPECPARTS/rpm-debuginfo.specpart
Processing files: valve-firmware-20231113.1-4.fc41.x86_64
Provides: valve-firmware = 20231113.1-4.fc41 valve-firmware(x86-64) = 20231113.1-4.fc41
Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
Requires: /usr/bin/bash libc.so.6()(64bit) libc.so.6(GLIBC_2.14)(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libc.so.6(GLIBC_2.3.4)(64bit) libc.so.6(GLIBC_2.33)(64bit) libc.so.6(GLIBC_2.38)(64bit) libc.so.6(GLIBC_2.4)(64bit) libgcc_s.so.1()(64bit) libgcc_s.so.1(GCC_3.0)(64bit) libgcc_s.so.1(GCC_3.3.1)(64bit) libm.so.6()(64bit) libm.so.6(GLIBC_2.2.5)(64bit) libm.so.6(GLIBC_2.27)(64bit) libm.so.6(GLIBC_2.29)(64bit) libstdc++.so.6()(64bit) libstdc++.so.6(CXXABI_1.3)(64bit) libstdc++.so.6(GLIBCXX_3.4)(64bit) libstdc++.so.6(GLIBCXX_3.4.15)(64bit) libstdc++.so.6(GLIBCXX_3.4.20)(64bit) libstdc++.so.6(GLIBCXX_3.4.21)(64bit) libstdc++.so.6(GLIBCXX_3.4.30)(64bit) rtld(GNU_HASH)
Processing files: valve-firmware-debugsource-20231113.1-4.fc41.x86_64

RPM build errors:
error: Empty %files file /builddir/build/BUILD/valve-firmware-20231113.1-build/valve-firmware-20231113.1/debugsourcefiles.list
    Empty %files file /builddir/build/BUILD/valve-firmware-20231113.1-build/valve-firmware-20231113.1/debugsourcefiles.list
Finish: rpmbuild valve-firmware-20231113.1-4.fc41.src.rpm
Finish: build phase for valve-firmware-20231113.1-4.fc41.src.rpm
INFO: chroot_scan: 1 files copied to /var/lib/copr-rpmbuild/results/chroot_scan
INFO: /var/lib/mock/fedora-41-x86_64-1726699240.972754/root/var/log/dnf5.log
ERROR: Exception(/var/lib/copr-rpmbuild/results/valve-firmware-20231113.1-4.fc41.src.rpm) Config(fedora-41-x86_64) 0 minutes 24 seconds
INFO: Results and/or logs in: /var/lib/copr-rpmbuild/results
INFO: Cleaning up build root ('cleanup_on_failure=True')
Start: clean chroot
INFO: unmounting tmpfs.
Finish: clean chroot
ERROR: Command failed: 
 # /usr/bin/systemd-nspawn -q -M 0ec5c426f7444dadb4bec53ba9b33808 -D /var/lib/mock/fedora-41-x86_64-1726699240.972754/root -a -u mockbuild --capability=cap_ipc_lock --rlimit=RLIMIT_NOFILE=10240 --capability=cap_ipc_lock --bind=/tmp/mock-resolv.1pq4oepz:/etc/resolv.conf --bind=/dev/btrfs-control --bind=/dev/mapper/control --bind=/dev/fuse --bind=/dev/loop-control --bind=/dev/loop0 --bind=/dev/loop1 --bind=/dev/loop2 --bind=/dev/loop3 --bind=/dev/loop4 --bind=/dev/loop5 --bind=/dev/loop6 --bind=/dev/loop7 --bind=/dev/loop8 --bind=/dev/loop9 --bind=/dev/loop10 --bind=/dev/loop11 --console=pipe --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin '--setenv=PROMPT_COMMAND=printf "\033]0;<mock-chroot>\007"' '--setenv=PS1=<mock-chroot> \s-\v\$ ' --setenv=LANG=C.UTF-8 --resolv-conf=off bash --login -c '/usr/bin/rpmbuild -bb  --target x86_64 --nodeps /builddir/build/originals/valve-firmware.spec'

Copr build error: Build failed
```

https://download.copr.fedorainfracloud.org/results/playtron/gaming/fedora-41-x86_64/08032576-valve-firmware/builder-live.log.gz

**Pull request merged**